src/main/kotlin/com/springernature/newversion/SummaryWriter.kt

```diff
@@ -0,0 +1,46 @@
+package com.springernature.newversion
+
+import java.io.File
+
+class SummaryWriter(val file: File) {
+
+    constructor(filename: String?) : this(
+        if (filename == null || filename.isBlank()) {
+            System.err.println("no filename for report provided, will not create a report")
+            File("/dev/null")
+        } else {
+            File(filename)
+        })
+
+    fun write(results: ChecksResult) {
+        writeHeader(file)
+        when (results) {
+            is FailedChecks -> {
+                writeFailures(results.errors)
+                writeSuccessfulUpdates(results.updates)
+            }
+            is SuccessfulChecks -> writeSuccessfulUpdates(results.updates)
+        }
+        writeFooter(file)
+    }
+
+    private fun writeHeader(file: File) {
+        file.writeText("# CF buildpack update action results\n")
+    }
+
+    private fun writeSuccessfulUpdates(updates: List<BuildpackUpdate>) {
+        file.appendText("\n## success\n\n")
+        updates.forEach { file.appendText("* currentBuildpack ${it.currentBuildpack} ${if (it.hasUpdate()) "has an update to " + it.latestUpdate else "has no update"}\n") }
+    }
+
+    private fun writeFailures(errors: Map<BuildpackUpdate, Exception>) {
+        file.appendText("\n## failures\n" +
```

Comment:
Might need to be conditionally written, only when updates.length > 0? Otherwise you have a heading without content?

Also: newline-ception